src/solvers/convex_bundle_method.jl

```diff
@@ -0,0 +1,586 @@
+* `k_max`:                     upper bound on the sectional curvature of the manifold.
+* `k_size`:                    (`100``) sample size for the estimation of the bounds on the sectional curvature of the manifold if `k_min`
+    and `k_max` are not provided.
+* `p_estimate`:                (`p`) the point around which to estimate the sectional curvature of the manifold.
+* `α`:                         (`(i) -> one(number_eltype(X)) / i`) a function for evaluating suitable stepsizes when obtaining candidate points at iteration `i`.
+* `ϱ`:                         curvature-dependent bound.
+* `evaluation`:                ([`AllocatingEvaluation`](@ref)) specify whether the subgradient works by
+   allocation (default) form `∂f(M, q)` or [`InplaceEvaluation`](@ref) in place, i.e. is
+   of the form `∂f!(M, X, p)`.
+* `inverse_retraction_method`: (`default_inverse_retraction_method(M, typeof(p))`) an inverse retraction method to use
+* `retraction`:                (`default_retraction_method(M, typeof(p))`) a `retraction(M, p, X)` to use.
+* `stopping_criterion`:        ([`StopWhenLagrangeMultiplierLess`](@ref)`(1e-8)`) a functor, see[`StoppingCriterion`](@ref), indicating when to stop
+* `vector_transport_method`:   (`default_vector_transport_method(M, typeof(p))`) a vector transport method to use
+* `sub_problem`:               a function evaluating with new allocations that solves the sub problem on `M` given the last serious iterate `p_last_serious`, the linearization errors `lin_errors`, and the transported subgradients `transported_subgradients`
+
+# Output
+
+the obtained (approximate) minimizer ``p^*``, see [`get_solver_return`](@ref) for details
+"""
+function convex_bundle_method(
+    M::AbstractManifold, f::TF, ∂f::TdF, p=rand(M); kwargs...
+) where {TF,TdF}
+    p_star = copy(M, p)
+    return convex_bundle_method!(M, f, ∂f, p_star; kwargs...)
+end
+@doc raw"""
+    convex_bundle_method!(M, f, ∂f, p)
+
+perform a bundle method ``p_{j+1} = \mathrm{retr}(p_k, -g_k)`` in place of `p`.
+
+# Input
+
+* `M`:  a manifold ``\mathcal M``
+* `f`:  a cost function ``f:\mathcal M→ℝ`` to minimize
+* `∂f`: the (sub)gradient ``∂f:\mathcal M→ T\mathcal M`` of F
+  restricted to always only returning one value/element from the subdifferential.
+  This function can be passed as an allocation function `(M, p) -> X` or
+  a mutating function `(M, X, p) -> X`, see `evaluation`.
+* `p`:  an initial value ``p_0=p ∈ \mathcal M``
+
+for more details and all optional parameters, see [`convex_bundle_method`](@ref).
+"""
+function convex_bundle_method!(
+    M::AbstractManifold,
+    f::TF,
+    ∂f!!::TdF,
+    p;
+    atol_λ::R=eps(),
+    atol_errors::R=eps(),
+    bundle_cap::Int=25,
+    diam::R=π / 3,# k_max -> k_max === nothing ? π/2 : (k_max ≤ zero(R) ? typemax(R) : π/3),
+    domain=(M, p) -> isfinite(f(M, p)),
+    m::R=1e-3,
+    k_min=nothing,
+    k_max=nothing,
+    k_size::Int=100,
+    p_estimate=nothing,
```

Comment:
This seems to be unused. there is a `p_estimate` passed to the state but that uses `p`. Can this be removed?